### PR TITLE
- created `Account` and `Transaction` entities

### PR DESCRIPTION
**Pull request by Likh-Alex**
- created `Repository` and `Service` layers for entities

### CODE REVIEW

**Review by Alex-Korzun — APPROVED** [review]

File: src/main/java/com/dev/bankingservice/entity/Account.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+@Data
+public class Account {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(name = "account_number", unique = true, nullable = false)
+    private String accountNumber;
+    @Enumerated(EnumType.STRING)
+    private Currency currency;
+    private Double balance;
```
Comment (boroda4436) [inline src/main/java/com/dev/bankingservice/entity/Account.java]: BigDecimal

File: src/main/java/com/dev/bankingservice/entity/Transaction.java [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @JoinColumn(name = "account_from", nullable = false)
+    @ManyToOne
+    private Account accountFrom;
+    @JoinColumn(name = "account_to", nullable = false)
+    @ManyToOne
+    private Account accountTo;
+    private Double amount;
```
Comment (boroda4436) [inline src/main/java/com/dev/bankingservice/entity/Transaction.java]:
```suggestion
    private BigDecimal amount;
```

File: src/main/java/com/dev/bankingservice/service/TransactionService.java [hunk 3]
```diff
@@ -0,0 +1,11 @@
+package com.dev.bankingservice.service;
+
+import com.dev.bankingservice.entity.Account;
+import com.dev.bankingservice.entity.Transaction;
+import java.util.List;
+
+public interface TransactionService {
+    List<Transaction> getAllByAccount(int page, int size, Account account);
+
+    void transferFunds(String accountFrom, String accountTo, Double amount);
```
Comment (boroda4436) [inline src/main/java/com/dev/bankingservice/service/TransactionService.java]:
```suggestion
    void transferFunds(String accountFrom, String accountTo, BigDecimal amount);
```

File: src/main/java/com/dev/bankingservice/service/impl/TransactionServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    private final AccountRepository accountRepository;
+
+    @Override
+    public List<Transaction> getAllByAccount(int page, int size, Account account) {
+        PageRequest pageable = PageRequest.of(page, size, Sort.by("date").descending());
+        return transactionRepository.getAllByAccount(account.getAccountNumber(), pageable);
+    }
+
+    @Override
+    public void transferFunds(String accountFrom, String accountTo, Double amount) {
```
Comment (boroda4436) [inline src/main/java/com/dev/bankingservice/service/impl/TransactionServiceImpl.java]: You can add `@Transactional` annotation for this method

**Review by boroda4436 — APPROVED** [review]

File: src/main/java/com/dev/bankingservice/entity/Account.java (hunk 1 above)
Comment (Likh-Alex) [reply src/main/java/com/dev/bankingservice/entity/Account.java]: Replaced

File: src/main/java/com/dev/bankingservice/entity/Transaction.java (hunk 2 above)
Comment (Likh-Alex) [reply src/main/java/com/dev/bankingservice/entity/Transaction.java]: Replaced

File: src/main/java/com/dev/bankingservice/service/TransactionService.java (hunk 3 above)
Comment (Likh-Alex) [reply src/main/java/com/dev/bankingservice/service/TransactionService.java]: Replaced

File: src/main/java/com/dev/bankingservice/service/impl/TransactionServiceImpl.java (hunk 4 above)
Comment (Likh-Alex) [reply src/main/java/com/dev/bankingservice/service/impl/TransactionServiceImpl.java]: Added annotation

**Pull request merged**